### PR TITLE
Bands RFC #1213

### PR DESCRIPTION
**Pull request by m-mohr**
**Related Issue(s):** #1213

**Proposed Changes:**

- `bands` is a new field in common metadata to replace `eo:bands` and `raster:bands` ([#1213](https://github.com/radiantearth/stac-spec/discussions/1213))
- The fields `data_type`, `nodata`, `statistics` and `unit` have been added to common metadata ([#1213](https://github.com/radiantearth/stac-spec/discussions/1213))

This PR needs to wait for the EO and Raster Extension PRs.

ToDo:
- https://github.com/stac-extensions/raster/issues/40
- https://github.com/stac-extensions/eo/issues/27
- https://github.com/stac-extensions/raster/pull/45
- https://github.com/stac-extensions/eo/pull/29

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md)
      **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec),
      and I have opened issue/PR #XXX to track the change.

### CODE REVIEW

File: item-spec/json-schema/common.json [hunk 1]
```diff
@@ -8,9 +8,15 @@
     {
       "$ref": "basics.json"
     },
+    {
+      "$ref": "bands.json"
+    },
```
Comment (emmanuelmathot) [inline item-spec/json-schema/common.json:13]: Isn't there a loop dependency here? Band is only part of the asset object right?

File: item-spec/common-metadata.md [hunk 2]
```diff
@@ -13,14 +13,9 @@ or [Collection Asset](../collection-spec/collection-spec.md#asset-object).
...
     - [Provider Object](#provider-object)
-      - [roles](#roles)
   - [Instrument](#instrument)
-    - [Additional Field Information](#additional-field-information)
-      - [platform](#platform)
-      - [instruments](#instruments)
-      - [constellation](#constellation)
-      - [mission](#mission)
-      - [gsd](#gsd)
+  - [Bands](#bands)
```
Comment (emmanuelmathot) [inline item-spec/common-metadata.md:17]: I still do not understand why bands is part of the common metadata and not only an child object for asset

Comment (m-mohr) [reply item-spec/common-metadata.md:17]: So that it can be used in all places.

File: item-spec/json-schema/common.json (hunk 1 above)
Comment (m-mohr) [reply item-spec/json-schema/common.json:13]: No, but it is recursive, indeed.

File: item-spec/common-metadata.md (hunk 2 above)
Comment (emmanuelmathot) [reply item-spec/common-metadata.md:17]: all right then I did not understand the outcome here: https://github.com/stac-extensions/raster/issues/22#issuecomment-1739782633
I thought dehydration was up to implementation and thus did not need the band at item level.

Comment (m-mohr) [reply item-spec/common-metadata.md:17]: In APIs, yes. But you still have static STACs and/or cases where you explicitly want it in Item Properties. That was my understanding.

**Comment by emmanuelmathot** [comment]
Concerning the last commit, there is no way to make suggestions on unchanged file so I simply committed it. Let me know.

**Comment by m-mohr** [comment]
We should discuss whether we want to explicitly disallow to use of "bands" for single-band use-cases where instead people must provide the properties at the asset/item properties level directly.

**Comment by emmanuelmathot** [comment]
> We should discuss whether we want to explicitly disallow to use of "bands" for single-band use-cases where instead people must provide the properties at the asset/item properties level directly.

I vote +1 for disallowing unique band

File: item-spec/common-metadata.md [hunk 3]
```diff
@@ -208,3 +203,131 @@ optical and short-wave IR bands are all 30 meters, but the panchromatic band is
...
+| ---------- | --------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| nodata     | number\|string                          | Values used to identify values that are nodata, either by the value as a number or as one of the following strings: `nan`, `inf` or `-inf`.    |
+| data_type  | string                                  | The data type of the values. One of the [data types as described below](#data-types).                                                          |
+| statistics | [Statistics Object](#statistics-object) | Statistics of all the values.                                                                                                                  |
+| unit       | string                                  | Unit of measurement of the value, preferably compliant to [UDUNITS-2](https://ncics.org/portfolio/other-resources/udunits2/) units (singular). |
```
Comment (m-mohr) [inline item-spec/common-metadata.md:292]: See also https://github.com/stac-extensions/raster/issues/47

**Review by emmanuelmathot — COMMENTED** [review]
@m-mohr Are we good for a merge in `dev`? At least to be able to release a beta version to validate extensions and pending PR that requires the band construct?

**Comment by m-mohr** [comment]
The extensions should validate without requiring 1.1. 

I think we could do the following:
- Finish and merge extensions (potentially changing the base branch temporarily to the latest stable version to not confuse readers)
- Release rc.1 versions
- Update and merge this PR
- Release 1.1.0-rc.1 version of STAC
- Prepare everything for 1.1.0 (especially examples)
- Let users check everything for a month or two
- Release new stable versions for everything

**Comment by m-mohr** [comment]
We should probably discuss in the PR when properties should be used on the asset level and when people should have a single-element bands array. I assume it's just if it's a named entity (i.e. `name` is required)?

**Comment by emmanuelmathot** [comment]
you mean allowing single band to be able to set properties only at band level?

**Comment by m-mohr** [comment]
I meant guiding users between the different options. 

### Single Band

For a single band asset, you could for example do one of the two options:

(1)
```json
{
  "assets": {
    "example": {
      "href": "example.tif",
      "bands": [
        {
          "data_type": "uint16",
          "eo:common_name": "red",
          "raster:spatial_resoltion": 10
        }
      ]
    }
  }
}
```

or

(2)
```json
{
  "assets": {
    "example": {
      "href": "example.tif",
      "data_type": "uint16",
      "eo:common_name": "red",
      "raster:spatial_resoltion": 10
    }
  }
}
```

If you add a name, it get's even more unclear:

(3)
```json
{
  "assets": {
    "example": {
      "href": "example.tif",
      "bands": [
        {
          "name": "r",
          "data_type": "uint16",
          "eo:common_name": "red",
          "raster:spatial_resoltion": 10
        }
      ]
    }
  }
}
```

Would you then do the following?

(4)
```json
{
  "assets": {
    "example": {
      "href": "example.tif",
      "data_type": "uint16",
      "eo:common_name": "red",
      "raster:spatial_resoltion": 10,
      "bands": [
        {
          "name": "r"
        }
      ]
    }
  }
}
```

### Multi-band

And then for multi-band files, you can often also de-duplicate some properties. 

Old example from 1.0:
(5)
```json
{
  "assets": {
    "example": {
      "href": "example.tif",
      "eo:bands": [
        {
          "name": "r",
          "common_name": "red"
        },
        {
          "name": "g",
          "common_name": "green"
        },
        {
          "name": "b",
          "common_name": "blue"
        }
      ],
      "raster:bands": [
        {
          "data_type": "uint16",
          "spatial_resolution": 10
        },
        {
          "data_type": "uint16",
          "spatial_resolution": 10
        },
        {
          "data_type": "uint16",
          "spatial_resolution": 10
        }
      ]
    }
  }
}
```

If you would just merge them, you end up with:
(6)
```json
{
  "assets": {
    "example": {
      "href": "example.tif",
      "bands": [
        {
          "name": "r",
          "eo:common_name": "red",
          "data_type": "uint16",
          "raster:spatial_resolution": 10
        },
        {
          "name": "g",
          "eo:common_name": "green",
          "data_type": "uint16",
          "raster:spatial_resolution": 10
        },
        {
          "name": "b",
          "eo:common_name": "blue",
          "data_type": "uint16",
          "raster:spatial_resolution": 10
        }
      ]
    }
  }
}
```

But ideally, I think, you could simplify to:
(7)
```json
{
  "assets": {
    "example": {
      "href": "example.tif",
      "data_type": "uint16",
      "raster:spatial_resolution": 10,
      "bands": [
        {
          "name": "r",
          "eo:common_name": "red"
        },
        {
          "name": "g",
          "eo:common_name": "green"
        },
        {
          "name": "b",
          "eo:common_name": "blue"
        }
      ]
    }
  }
}
```

I think the difference between 5 and 7 shows the beauty of the new concept nicely ;-)

**Review by emmanuelmathot — APPROVED** [review]

File: best-practices.md [hunk 4]
```diff
@@ -428,6 +422,236 @@ If your data for the Item does not come with a thumbnail already we do recommend
 | Spatial extent                 | Limited                                             | Full                                                               | Full                                                                                         |
 | Use case                       | Quick overview, often in lists of items/collections | Display for a single Item/Collection, sometimes shown on a web map | Display for a single Item/Collection, often shown on a map, may be displayed in GIS software |
 
+### Bands
+
+The new `bands` array in combination with the property inheritance introduced in STAC 1.1 provides users with more flexibility.
```
Comment (gadomski) [inline best-practices.md:427]: It's new now, but it won't always be.

```suggestion
As of STAC 1.1, the `bands` array can be used in combination with property inheritance to provide users with more flexibility.
```

**Review by gadomski — APPROVED** [review]

**Pull request merged**